tools/delete_imaging_upload.pl

```diff
@@ -0,0 +1,1628 @@
+ C<mri_violations_log> and C<FullPath> => absolute path of the MINC file.
+
+=cut
+sub getMriViolationsLogFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath, $scanTypesToDeleteRef) = @_;
+    
+    return [] if $tarchiveID eq 'NULL';
+
+    my $query;
+    if(@$scanTypesToDeleteRef) {
+        $query = 'SELECT LogID, mvl.MincFile '
+               . 'FROM mri_violations_log mvl '
+               . 'JOIN files f ON (f.File COLLATE utf8_bin = mvl.MincFile) '
+               . 'JOIN mri_scan_type mst ON (f.AcquisitionProtocolID = mst.ID) '
+               . 'WHERE TarchiveID = ? '
+               . 'AND mst.Scan_type IN('
+               . join(',', ('?') x @$scanTypesToDeleteRef)
+               . ')';
+    } else {
+        $query = 'SELECT LogID, MincFile FROM mri_violations_log WHERE TarchiveID = ?';
+    }
+    my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID, @$scanTypesToDeleteRef);
+
+    # Set full path of every file
+    foreach(@$filesRef) {
+        $_->{'FullPath'} = $_->{'MincFile'} =~ /^\// 
+            ? $_->{'MincFile'} : "$dataDirBasePath/$_->{'MincFile'}";
+    } 
+
+    return $filesRef;   
+}
+
+=pod
+
+=head3 getMRICandidateErrorsFilesRef($dbh, $tarchiveID, $dataDirBasePath, $scanTypeToDeleteRef)
+
+Get the absolute paths of all the files associated to a DICOM archive that are listed in 
+table C<MRICandidateErrors>.
+
+INPUTS:
+  - $dbh   : database handle reference.
+  - $tarchiveID: ID of the DICOM archive.
+  - $dataDirBasePath: config value of setting C<dataDirBasePath>.
+  - $scanTypesToDeleteRef: reference to the array that contains the list of scan type names to delete.
+
+RETURNS: 
+ - an array of hash references. Each hash has three keys: C<ID> => ID of the record in the 
+   table, C<MincFile> => value of column C<MincFile> for the MINC file found in table 
+   C<MRICandidateErrors> and C<FullPath> => absolute path of the MINC file.
+
+=cut
+sub getMRICandidateErrorsFilesRef {
+    my($dbh, $tarchiveID, $dataDirBasePath, $scanTypeToDeleteRef) = @_;
+    
+    return [] if $tarchiveID eq 'NULL' || @$scanTypeToDeleteRef;
+
+    # Get file path of each files in files directly tied
+    # to $tarchiveId
+    $query = 'SELECT ID, MincFile FROM MRICandidateErrors WHERE TarchiveID = ?';
+    my $filesRef = $dbh->selectall_arrayref($query, { Slice => {} }, $tarchiveID);
+
+    # Set full path of every file
+    foreach(@$filesRef) {
+        $_->{'FullPath'} = $_->{'MincFile'} =~ /^\// 
+            ? $_->{'MincFile'} : "$dataDirBasePath/$_->{'MincFile'}";
+    } 
+    
+    return $filesRef;   
+}
+
+=pod
+
+=head3 getBackupFileName($tarchiveID)
+
+Gets the name of the backup compressed file that will contain a copy of all the files
+that the script will delete.
+
+INPUTS:
+  - $tarchiveID: ID of the DICOM archive (in table C<tarchive>) associated to the upload(s) passed on the command line.
+
+getbackup
```

Comment:
not sure why getbackup is here? Maybe forgotten to remove when added the doc?

Reply:
A zombie subroutine....Gone.